py/jupyterlite/pyproject.toml

```diff
@@ -9,9 +9,7 @@ authors = [
 ]
 dependencies = [
     "jupyterlite-core",
-    # TODO
-    # "jupyterlite-core[lab]",
-    # "jupyterlite-pyodide-kernel",
+    "jupyterlite-pyodide-kernel >=0.0.4",
```

Comment:
why not also the compatible lab?

Reply:
I guess we could indeed include more dependencies as mentioned in https://github.com/jupyterlite/jupyterlite/issues/993 (like ipywidgets). Maybe we can first see how it goes and add them later.

Reply:
My thinking here: the "happy path" of one of the most complex dependency issues, relying on the users' package manager of choice to make `sys.prefix` whole, will start getting less happy as extensions don't install/solve correctly because of jupyterlab 4... and `[extras]` are basically broken. 

A future `jupyterlab-core` can then depend on `jupyterlab >=4,<4.1`, even when we maintain the `3.5` compatible one.

I like widgets for this same reason: we can support `ipywidgets >=7.6,<8`, but there's no guarantee all future versions will have compatible `jupyterlab_widgets` releases, and getting kernels up to supporting widgets represents a _huge_ amount of effort (right next to arbitrary python packages, and iteratively reducing friction there).

Reply:
Let's track that in a separate issue about which package to include in `jupyterlite` by default.

Reply:
Opened https://github.com/jupyterlite/jupyterlite/issues/1003